src/missing_prices.py

```diff
@@ -128,14 +135,14 @@ def run_missing_prices() -> None:
     print(f"Fetched {len(tokens)} traded tokens from Dune without prices")
     found, res = 0, []
     for token in tokens:
-        if token["address"].lower() in coins:
-            paprika_data = coins[token["address"].lower()]
+        if token["address"] in coins:
```

Comment:
If these tokens are now object instances, shouldn't we be able to reference them as `token.address`?

Reply:
`Schema.load` doesn't automatically turn it into an instance of an object. For that to happen we would have to extend the functionality by adding:

```python
class TokenSchema(Schema):
    """TokenSchema CoinSchema for serializing/deserializing token data"""

    address = EthereumAddress(required=True)
    decimals = fields.Int(required=True)
    popularity = fields.Int()
    symbol = fields.String()

    @post_load
    def make_user(self, data, **kwargs):
        return Token(**data)


@dataclass
class Token:
    address: Address
    decimals: int
    symbol: str
    popularity: int
```

Reply:
That seems a bit like double work. Or rather the same amount of work as having a Token.from_dict or __init__ method. Doesn't it?

Reply:
That's correct.

I think for this to work properly we should validate the data from the API using marshmallow only and not iterate over it manually. 

But then you're using marshmallow to filter data which I don't think it's intended to do.

Unfortunately I don't think I have enough experience with marshmallow to give you a concrete answer to which is better. What's your opinion on that?

Reply:
Send give for now. We will gain experience using it once it's in here.